client/dashboard/app/queries/site-sftp.ts

```diff
@@ -1,16 +1,22 @@
+import { mutationOptions, queryOptions } from '@tanstack/react-query';
 import { fetchSftpUsers, createSftpUser, resetSftpPassword } from '../../data/site-hosting-sftp';
 import { queryClient } from '../query-client';
 import type { SftpUser } from '../../data/site-hosting-sftp';
 
-export const siteSftpUsersQuery = ( siteId: number ) => ( {
-	queryKey: [ 'site', siteId, 'sftp-users' ],
-	queryFn: () => fetchSftpUsers( siteId ),
-	meta: {
-		persist: false,
-	},
-} );
+export const siteSftpUsersQuery = ( siteId: number ) =>
+	queryOptions( {
+		queryKey: [ 'site', siteId, 'sftp-users' ],
+		queryFn: () => fetchSftpUsers( siteId ),
+		meta: {
+			persist: false,
+		},
+	} );
+
+const updateCurrentSftpUsers = ( currentSftpUsers: SftpUser[] | undefined, sftpUser: SftpUser ) => {
+	if ( ! currentSftpUsers ) {
+		return [ sftpUser ];
+	}
```

Comment:
Another similar bug fix.
If sftp users updated before they are ever been queried, then this would have tried to call `findIndex()` on `undefined`.

To fix the issue, I decided that if the sftp user is updated before it is ever queried, then we can just put this one new user in the cache.